scene/3d/ccd_ik_3d.cpp

```diff
@@ -31,18 +31,26 @@
 #include "ccd_ik_3d.h"
 
 void CCDIK3D::_process_joints(double p_delta, Skeleton3D *p_skeleton, Vector<ManyBoneIK3DJointSetting *> &p_joints) {
-	// Solve IK here in extended class. Show example for iterating parent to child below.
-	/*
+	// Solve IK here in extended class.
+	// Show example for iterating parent to child below.
 	for (int i = 0; i < p_joints.size(); i++) {
 		ManyBoneIK3DSolverInfo *solver_info = p_joints[i]->solver_info;
 		if (!solver_info) {
 			continue; // Means not extended end bone.
 		}
-		Vector3 destination;
+		Vector<Transform3D> targets;
+		Vector<Transform3D> moveds;
```

Comment:
Why is it Vector? Do you need to pass all joints to constraint?

My thoughts: IK does not consider constraint, it calculates one rotation by the relationship with the next index joint here. After that constraint only needs to be passed the previous and following transforms.

Reply:
For positional / rotational / positional + rotational constraint it is required that the root bone plus all the bones are passed in the order specified in many bone ik design document.

Reply:
Root bones can be have translation. All other bones cannot.

Reply:
I’ll ask eron if this is correct. Wait a bit.

Reply:
From Eron

> That only works if you only have 1 target effector pair being solved for. 
> 
> But your torso often needs to rotate to account for what you want both of your hands to be.
> 
> In this case N= 2.
> 
> If you want to keep your head in the same place as well, N=3